misk-service/src/main/kotlin/misk/ServiceModule.kt

```diff
@@ -125,6 +133,11 @@ class ServiceModule constructor(
   inline fun <reified T : Service> dependsOn(qualifier: KClass<out Annotation>? = null) =
     dependsOn(T::class.toKey(qualifier))
 
+  inline fun <reified T : Service> dependsOn(
+    qualifier: KClass<out Annotation>? = null,
+    condition: Boolean,
```

Comment:
I wonder if accepting a lambda instead like `() -> Boolean` would be more flexible for inline expressions.